### ISSUE TITLE
Add edge-cases support in Pokemon Showdown pokedex reader

### ISSUE THREAD

**Issue by mirekdlugosz** [mirekdlugosz/create-pokemon-team#29]
Once #26 is done, extend Pokemon Showdown pokedex reader class to support edge cases:

* moves changing type depending on Pokemon type, item, terrain or weather (e.g. `naturalgift`, `judgment`, `multiattack`, `revelationdance`)
* Pokemon with different forms (Meowstic Male and Female)
* Pokemon freely changing forms (Arceus, Genesect, Hoopa etc.)
* Mega evolutions
* Regional variants (Alola - pay special attention to evolution lines, as sometimes previous evolution is Alolan and sometimes it is normal)
* data to ignore (see `_ignore_pokemon` in Eeeveedex reader - Totem Pokemon, Greninja forms, Cosplay Pikachu etc.)

**Comment by mirekdlugosz** [comment]
Peculiar edge case: there are Pokemon with multiple forms, where each form has different moveset, but Pokemon can change form at the whim. Moves from one form are available to another form. But there is a catch - some forms are not available in all games in generation.

* Deoxys in gen 3
  * ruby-sapphire has only "deoxys" (normal)
  * emerald has only "deoxysspeed"
  * firered-leafgreen has only "deoxysattack" and "deoxysdefense" (technically, Fire Red has Attack, while Leaf Green has Defense, but we always consider games in pairs)
* Giratina in gen 4
  * diamond-pearl has only "giratina"
  * platinum has both "giratina" and "giratinaorigin"
* Shaymin in gen 4
  * diamond-pearl has only "shaymin"
  * platinum has both "shaymin" and shayminsky"

**Comment by mirekdlugosz** [comment]
Mostly done in branch [showdown-full](https://github.com/mirekdlugosz/create-pokemon-team/tree/showdown-full), which was rebased and merged into master as commits 0c44227c8c692e0c424171a11341962579650384..7d16e75aca98f0f18e96c97130a2544a62975316. Remaining issues require changes to pokedex class and constants, and they will break veekun reader - so they will need to be done as part of #27 and/or #30.
Also, exceptions from previous comment are not handled properly. This has now new issue #38.
